charon/src/cli_options.rs

```diff
@@ -98,7 +98,7 @@ performs: `y := (x as E2).1`). Producing a better reconstruction is non-trivial.
     /// This is for Nix: outside of Nix, we use Rustup to call the proper version
     /// of Cargo (and thus need this argument), but within Nix we build and call a very
     /// specific version of Cargo.
-    #[structopt(long = "cargo-no-rust-version")]
+    #[structopt(long = "cargo-no-rust-version", env = "CARGO_NO_RUST_VERSION")]
```

Comment:
The CI is broken and I think this is because of this line: now we can't just use `--cargo-no-rust-version`, we actually have to provide an argument:
```
nix build .#checks.x86_64-linux.tests --show-trace -L
warning: Git tree '/home/soho/Projects/prosecco/charon-main' is dirty
warning: input 'crane' has an override for a non-existent input 'flake-utils'
tests> unpacking sources
tests> unpacking source archive /nix/store/v742lh8q0bz1n7nwgrib5qx38a1njzsy-tests
tests> source root is tests
tests> patching sources
tests> Executing configureCargoCommonVars
tests> decompressing cargo artifacts from /nix/store/5217lfw161v66ivgblhg3r4hzlg3g35i-tests-deps-0.1.0/target.tar.zst to target
tests> updateAutotoolsGnuConfigScriptsPhase
tests> configuring
tests> will append /build/tests/.cargo-home/config.toml with contents of /nix/store/j11cab5rr84w737ph9m0gn1lxv73xa8v-vendor-cargo-deps/config.toml
tests> default configurePhase, nothing to do
tests> building
tests> /nix/store/7hgd6qbm0yrlm0nj81ckcjilpmcscc84-charon-0.1.0/bin/charon --cargo-no-rust-version --crate nested_borrows --input src/nested_borrows.rs --no-code-duplication --dest /nix/store/l7x9hy93lls5bx20pm8c8fwx0zxgn2ls-tests/llbc
tests> error: The argument '--cargo-no-rust-version <cargo-no-rust-version>' requires a value but none was supplied
tests> USAGE:
tests>     charon --cargo-no-rust-version <cargo-no-rust-version> --crate <crate-name> --dest <dest-dir> --input <input-file> --no-code-duplication
tests> For more information try --help
tests> make: *** [Makefile:75: test-nested_borrows] Error 1
error: builder for '/nix/store/qaannfz69pxhv03hd4y65b9wjvv7d6a4-tests.drv' failed with exit code 2;
       last 10 log lines:
       > default configurePhase, nothing to do
       > building
       > /nix/store/7hgd6qbm0yrlm0nj81ckcjilpmcscc84-charon-0.1.0/bin/charon --cargo-no-rust-version --crate nested_borrows --input src/nested_borrows.rs --no-code-duplication --dest /nix/store/l7x9hy93lls5bx20pm8c8fwx0zxgn2ls-tests/llbc
       > error: The argument '--cargo-no-rust-version <cargo-no-rust-version>' requires a value but none was supplied
       >
       > USAGE:
       >     charon --cargo-no-rust-version <cargo-no-rust-version> --crate <crate-name> --dest <dest-dir> --input <input-file> --no-code-duplication
       >
       > For more information try --help
       > make: *** [Makefile:75: test-nested_borrows] Error 1
       For full logs, run 'nix log /nix/store/qaannfz69pxhv03hd4y65b9wjvv7d6a4-tests.drv'.
Makefile:121: recipe for target 'nix-tests' failed
make: *** [nix-tests] Error 1
```